pf4j/src/main/java/org/pf4j/PluginClassLoader.java

```diff
@@ -15,38 +15,35 @@
  */
 package org.pf4j;
 
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
 import java.io.File;
 import java.io.IOException;
 import java.net.URL;
 import java.net.URLClassLoader;
-import java.util.ArrayList;
-import java.util.Collection;
-import java.util.Collections;
-import java.util.Enumeration;
-import java.util.List;
-import java.util.Objects;
-
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
+import java.util.*;
 
 /**
  * One instance of this class should be created by plugin manager for every available plug-in.
  * By default, this class loader is a Parent Last ClassLoader - it loads the classes from the plugin's jars
  * before delegating to the parent class loader.
- * Use {@link #parentFirst} to change the loading strategy.
+ * Use {@link #classLoadingOrder} to change the loading strategy.
  *
  * @author Decebal Suiu
  */
 public class PluginClassLoader extends URLClassLoader {
 
     private static final Logger log = LoggerFactory.getLogger(PluginClassLoader.class);
+    private static final SortedSet<ClassLoadingSource> DEFAULT_ORDER = new TreeSet<>(Arrays.asList(ClassLoadingSource.PLUGIN, ClassLoadingSource.DEPENDENCIES, ClassLoadingSource.PARENT));
```

Comment:
I think that it's better to create a `ClassLoadingStrategy` and to move there these constants and to move `ClassLoadingSource` in `ClassLoadingStrategy` as a inner. 
What do you think?

Reply:
I don't think that we need a `SortedSet`. A `List` implementation (`ArrayList`) is enough.  The `ClassLoadingStrategy`  will contains a method `getSources(): Iterable`.

Reply:
@decebals this was to prevent the case where ClassLoadingSource is being added twice by users who use:
`    public PluginClassLoader(PluginManager pluginManager, PluginDescriptor pluginDescriptor, ClassLoader parent, SortedSet<ClassLoadingSource> classLoadingOrder) {`

Reply:
i think the proper name for ClassLoadingStrategy , should be something like DefaultClassLoadingStrategy to express that these are the default options and are not exhaustive ( though its quite long in name) what do you say ?

Reply:
also , regarding making it inner class  - since logically ClassLoadingStrategy  is actually DefaultClassLoadingStrategy  it means users using the constructor above will have to use the enums as
` DefaultClassLoadingStrategy.ClassLoadingStrategy.Plugin`
which seems a bit long , so summing up what iu think is good to do here is create a class called DefaultClassLoadingStrategy   which contains the two constants , but keep ClassLoadingStrategy as a regular non inner class

Reply:
> i think the proper name for ClassLoadingStrategy , should be something like DefaultClassLoadingStrategy to express that these are the default options and are not exhaustive ( though its quite long in name) what do you say ?

We have only 6 possible combination no? I think that we can create all these combination and make the constructor of class private.

I think something like this:
- `A` = from Application source
- `P` = from Plugin source
- `D` = from Dependency source

Possible combination:

```
ClassLoadingStrategy.APD 
ClassLoadingStrategy.ADP
ClassLoadingStrategy.PAD
ClassLoadingStrategy.DAP
ClassLoadingStrategy.DPA
ClassLoadingStrategy.PDA
```

Do you think that is more clear?

Reply:
i have done t the way you have suggested , the only issue that might happen is if we add another loading source for some reason we will now have 2^4 options which is not maintainable , but since i dont see another loading source ever being added here i think its ok , if for some unforeseen reason there will be another loading source ClassLoadingStrategy constructor should become public